src/main/kotlin/org/rust/cargo/toolchain/wsl/Utils.kt

```diff
@@ -14,6 +18,26 @@ fun WSLDistribution.expandUserHome(path: String): String {
     return "$userHome${path.substring(1)}"
 }
 
-fun Path.hasExecutableOnWsl(toolName: String): Boolean = pathToExecutableOnWsl(toolName).toFile().isFile
+fun Path.hasExecutableOnWsl(toolName: String): Boolean {
+    checkEdtAndReadAction()
+    return pathToExecutableOnWsl(toolName).toFile().isFile
+}
 
 fun Path.pathToExecutableOnWsl(toolName: String): Path = resolve(toolName)
+
+private val LOG: Logger = Logger.getInstance("org.rust.cargo.toolchain.Utils")
+
+private val REPORTED_STACK_TRACES: MutableSet<String> = ContainerUtil.newConcurrentSet()
+
+private fun checkEdtAndReadAction() {
+    val application = ApplicationManager.getApplication() ?: return
+    if (!application.isInternal || application.isHeadlessEnvironment) return
+    val message = when {
+        application.isDispatchThread -> "Access to WSL filesystem on EDT"
+        application.isReadAccessAllowed -> "Access to WSL filesystem under ReadAction"
```

Comment:
Let's add `@NonNls` annotation here to mark these hardcoded literals as non intended to be localized